include/linker/utils.h

```diff
@@ -57,4 +57,34 @@ static inline bool linker_is_in_rodata(const void *addr)
 	#undef RO_END
 }
 
+/**
+ * @brief Check if address is in log string section.
+ *
+ * Note that this may return false if the address lies outside
+ * the compiler's __log_strings_section
+ *
+ * @param addr Address.
+ *
+ * @return True if address identified within __log_strings_section section.
+ */
+#if defined(CONFIG_LOG2_FMT_SECTION)
```

Comment:
imo this ifdef is not necessary. Section is created always. It's just it will be empty when CONFIG_LOG2_FMT_SECTION is not set.